optaplanner-examples/src/test/java/org/optaplanner/examples/integrationtests/entitylowcount/LowEntityCountTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.integrationtests.entitylowcount;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.config.SolverConfigContext;
+import org.optaplanner.core.config.constructionheuristic.ConstructionHeuristicPhaseConfig;
+import org.optaplanner.core.config.constructionheuristic.ConstructionHeuristicType;
+import org.optaplanner.core.config.heuristic.selector.move.composite.UnionMoveSelectorConfig;
+import org.optaplanner.core.config.heuristic.selector.move.generic.ChangeMoveSelectorConfig;
+import org.optaplanner.core.config.heuristic.selector.move.generic.SwapMoveSelectorConfig;
+import org.optaplanner.core.config.heuristic.selector.move.generic.chained.TailChainSwapMoveSelectorConfig;
+import org.optaplanner.core.config.localsearch.LocalSearchPhaseConfig;
+import org.optaplanner.core.config.localsearch.decider.acceptor.AcceptorConfig;
+import org.optaplanner.core.config.localsearch.decider.forager.LocalSearchForagerConfig;
+import org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig;
+import org.optaplanner.core.config.solver.SolverConfig;
+import org.optaplanner.core.config.solver.termination.TerminationConfig;
+import org.optaplanner.examples.common.app.DataSetLoader;
+import org.optaplanner.examples.common.app.SolverBuilder;
+import org.optaplanner.examples.tsp.app.TspApp;
+import org.optaplanner.examples.tsp.domain.TspSolution;
+import org.optaplanner.examples.tsp.domain.Visit;
+import org.optaplanner.examples.tsp.solver.score.TspIncrementalScoreCalculator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalStateException;
+
+public class LowEntityCountTest {
```

Comment:
Naming is very important. Given there a few thousands class in optaplanner, it needs to fit.

1) it is TSP example specific, so it should start with "TSP".
2) It is similar to NQueensConstructionHeuristicTrackingTest: a test writing in a specific example.
So for consistency, **the package should be org.optaplanner.examples.tsp.solver**
3) It tests the solver, so it should end with SolverTest. Which brings us to something like TspFewVisitSolverTest.

But...

Reply:
Take a look at DefaultSolverTest, which has solveStopsWhenUninitialized() and solveStopsWhenPartiallyInitialized().

Do we really need all this complex TSP example to test if the solver behaves correctly with few entities?
We have TestdataChainedSolution (= the simplest form of TSP), which is in the test classes of optaplanner-core.

**Would it be possible to add a zeroEntity() test method on DefaultSolverTest that uses TestdataChainedSolution?**

Reply:
@ge0ffrey we've already had a discussion about package naming in this PR. This is about a point of view; the test is not TSP specific, the test focuses on behaviour of OptaPlanner on very small datasets. And it happens to use TSP for the purpose.

There are going to be more tests like that, as the original QE tests suite was build around existing examples. Different tests there rely on different examples depending on their appropriateness for that particular test.

We agreed with @yurloc that introducing a new package called "integrationtests" would be the best option. The motivation is to make it clear these are not regular tests used to verify each examples, but integration tests that might depends on various examples.